src/object/obj_tx.c

```diff
@@ -355,6 +365,18 @@ dc_tx_alloc(daos_handle_t coh, daos_epoch_t epoch, uint64_t flags,
 	*ptx = tx;
 
 	return 0;
+out:
+	dc_cont_put(dc);
+
+	if (tx) {
+		if (tx->tx_pool)
+			dc_pool_put(tx->tx_pool);
+		if (tx->tx_req_cache)
```

Comment:
no need to check if !NULL for calling D_FREE

Reply:
Fix